services/sky/document_view.cc

```diff
@@ -130,9 +130,10 @@ void DocumentView::OnEmbed(
   services_provided_by_embedder_ = services_provided_by_embedder.Pass();
 
   Load(response_.Pass());
+}
 
+void DocumentView::UpdateRootMetricsAndAddObserver() {
   UpdateRootSizeAndViewportMetrics(root_->bounds());
-
   root_->AddObserver(this);
```

Comment:
We should be able to add the observer right away.  It's only UpdateRootSizeAndViewportMetrics that we need to delay.